<launch>
	<env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find simple_controller)/config/rosconsole.config"/>
    <arg name="control_velocity" default="true"/>
    <arg name="rqt_persp" default="$(find simple_controller)/cfg/steer_error.perspective"/>
    <arg name="velocity_noise" default="0.0"/>
    <!-- start cart in stage world -->
    <include file="$(find cart_launch)/launch/cart_stage.launch">
        <arg name="control_velocity" value="$(arg control_velocity)"/>
        <arg name="velocity_noise" value="$(arg velocity_noise)"/>
    </include>
    <!-- start simple controller -->
    <node name="controller" pkg="simple_controller" type="controller_node" output="screen">
    	<rosparam command="load" file="$(find simple_controller)/launch/controller.yaml" ns="simple_controller" />
    	<remap from="/controller/simple_controller/ground_truth" to="/robot/base_pose_ground_truth"/>
    	<remap from="/controller/simple_controller/odom" to="/robot/odom"/>
    	<remap from="steering" to="/robot/steering"/>
    </node>
    <!-- start sending velocity command -->
    <!-- node name="vel_node" pkg="rostopic" type="rostopic" args="pub /robot/velocity std_msgs/Float32  4.0 -r10">
    </node-->
   <!-- start rqt via script -->
    <node name="rqt" pkg="simple_controller" type="start_rqt" args="--perspective-file $(arg rqt_persp)" output="log" if="$(eval rqt_persp != '')">
    </node>

</launch>
